src/v/cloud_storage/segment_chunk_data_source.cc

```diff
@@ -93,9 +93,9 @@ ss::future<> chunk_data_source_impl::load_stream_for_chunk(
     vlog(_ctxlog.debug, "loading stream for chunk starting at {}", chunk_start);
 
     co_await load_chunk_handle(chunk_start)
-      .handle_exception([this](const std::exception_ptr& ex) {
-          return maybe_close_stream().then(
-            [&ex] { std::rethrow_exception(ex); });
+      .handle_exception([this](std::exception_ptr ex) -> ss::future<> {
+          co_await maybe_close_stream();
```

Comment:
coroutine lambdas are rejected by clang-tidy. i think the original version but making a copy of `ex` would be sufficient, right?

Reply:
Vlad ended up writing this with coroutines: https://github.com/redpanda-data/redpanda/pull/11383

Reply:
oh got it. thanks for flagging. ill close this pr